MetadataProcessor.Shared/Utility/nanoBitmapProcessor.cs

```diff
@@ -22,59 +23,69 @@ internal sealed class nanoBitmapProcessor
             writer.WriteUInt32((uint)_bitmap.Width);
             writer.WriteUInt32((uint)_bitmap.Height);
 
             writer.WriteUInt16(0x00);   // flags
 
             var nanoImageFormat = GetnanoImageFormat(_bitmap.RawFormat);
 
-            if (nanoImageFormat != 0)
+            if (nanoImageFormat != 0)  // For GIF and JPEG, we do not convert 
             {
                 writer.WriteByte(0x01);     // bpp
                 writer.WriteByte(nanoImageFormat);
                 _bitmap.Save(writer.BaseStream, _bitmap.RawFormat);
             }
             else
             {
-                writer.WriteByte(0x10);     // bpp
+                byte bitsPerPixel = 16;
+                writer.WriteByte(bitsPerPixel);     // bpp
                 writer.WriteByte(nanoImageFormat);
 
-                var rect = new Rectangle(Point.Empty, _bitmap.Size);
-                using (var convertedBitmap =
-                    _bitmap.Clone(new Rectangle(Point.Empty, _bitmap.Size),
-                        PixelFormat.Format16bppRgb565))
+                try
                 {
-                    var bitmapData = convertedBitmap.LockBits(
-                        rect, ImageLockMode.ReadOnly, convertedBitmap.PixelFormat);
+                    Bitmap clone = new Bitmap(_bitmap.Width, _bitmap.Height, PixelFormat.Format16bppRgb565);
+                    using (Graphics gr = Graphics.FromImage(clone))
+                {
+                        gr.DrawImageUnscaled(_bitmap, 0, 0);
+                    }
 
-                    var buffer = new short[bitmapData.Stride * convertedBitmap.Height / sizeof(short)];
-                    System.Runtime.InteropServices.Marshal.Copy(
-                        bitmapData.Scan0, buffer, 0, buffer.Length);
+                    Rectangle rect = new Rectangle(0, 0, clone.Width, clone.Height);
+                    BitmapData bitmapData = clone.LockBits(rect, ImageLockMode.ReadOnly, PixelFormat.Format16bppRgb565);
+
+                    byte[] data = new byte[clone.Width * clone.Height * 2]; // 2 bytes per pixel
+
+                    System.Runtime.InteropServices.Marshal.Copy(bitmapData.Scan0, data, 0, data.Length);
+                    clone.UnlockBits(bitmapData);
+                    writer.WriteBytes(data);
 
-                    convertedBitmap.UnlockBits(bitmapData);
-                    foreach (var item in buffer)
-                    {
-                        writer.WriteInt16(item);
-                    }
                 }
+                catch
+                    {
+                    throw new NotSupportedException($"PixelFormat ({_bitmap.PixelFormat.ToString()}) could not be converted to Format16bppRgb565.");
+                };
             }
         }
 
         private byte GetnanoImageFormat(
             ImageFormat rawFormat)
         {
-            if (rawFormat.Equals(ImageFormat.Gif))
+            if (rawFormat.Equals(ImageFormat.Bmp)) // Native == byte c_TypeBitmap = 0;
```

Comment:
Please improve these comments here too.
Also isn't prefereable to just return the enum and cast to byte?